vm/src/obj/objbyteinner.rs

```diff
@@ -1202,6 +1326,23 @@ pub fn try_as_byte(obj: &PyObjectRef) -> Option<Vec<u8>> {
     })
 }
 
+#[inline]
+fn count_substring(haystack: &[u8], needle: &[u8], maxcount: Option<usize>) -> usize {
+    let substrings = haystack.find_iter(needle);
+    if let Some(maxcount) = maxcount {
+        let mut count = 0;
```

Comment:
How about `min(substring.count(), maxcount)`?

Reply:
When `substring.count()` is big and `maxcount` is small, the iteration cost will be different by early break.

Reply:
Oh, I used it with `take()` and it works. Thanks!